### DISCUSSION TITLE
Could I copy files in background?

### DISCUSSION THREAD

**Discussion by dennys** [WinMerge/winmerge#2172] [Q&A] [answered]
When I compare 2 directories and copy files, it runs the copy job in the foreground and I cannot operate Winmerge before the job is completed.
Is it possible to run the copy job in the background?

**Comment by sdottaka** [comment]
Unfortunately, it is currently not possible to copy files in the background.

**Reply by dennys** [reply]
Got it... I'm not sure about the possibility to add this feature, could I issue a feature request?
I'm not familiar with C/C++, I'm a Java programmer. It seems C++ can use CopyFileEx to copy files,
 but it seems not easy to run it in the background?

**Reply by sdottaka** [reply, marked as answer]
To achieve this, I think it is necessary to create a new thread, have that thread copy the files, and when it finishes, notify the main thread and reflect the results.

**Reply by dennys** [reply]
Thanks for your reply, I created an issue for this feature request. https://github.com/WinMerge/winmerge/issues/2173
Please consider adding it, thanks.
